addons/product/models/product.py

```diff
@@ -719,17 +719,47 @@ def toggle_active(self):
 class ProductPackaging(models.Model):
     _name = "product.packaging"
     _description = "Product Packaging"
-    _order = 'sequence'
+    _order = 'product_id, sequence, id'
     _check_company_auto = True
 
-    name = fields.Char('Package Type', required=True)
+    name = fields.Char('Product Packaging', required=True)
     sequence = fields.Integer('Sequence', default=1, help="The first in the sequence is the default one.")
     product_id = fields.Many2one('product.product', string='Product', check_company=True)
-    qty = fields.Float('Contained Quantity', help="Quantity of products contained in the packaging.")
+    qty = fields.Float('Contained Quantity', default=1, help="Quantity of products contained in the packaging.")
     barcode = fields.Char('Barcode', copy=False, help="Barcode used for packaging identification. Scan this packaging barcode from a transfer in the Barcode app to move all the contained units")
     product_uom_id = fields.Many2one('uom.uom', related='product_id.uom_id', readonly=True)
     company_id = fields.Many2one('res.company', 'Company', index=True)
 
+    _sql_constraints = [
+        ('positive_qty', 'CHECK(qty > 0)', 'Contained Quantity should be positive.')
+    ]
+
+    def _check_qty(self, product_qty, uom_id, rounding_method="HALF-UP"):
+        """Check if product_qty in given uom is a multiple of the packaging qty.
+        If not, rounding the product_qty to closest multiple of the packaging qty
+        according to the rounding_method "UP", "HALF-UP or "DOWN".
+        """
+        default_uom = self.product_id.uom_id
+        packaging_qty = default_uom._compute_quantity(self.qty, uom_id)
+        # We do not use the modulo operator to check if qty is a mltiple of q. Indeed the quantity
+        # per package might be a float, leading to incorrect results. For example:
+        # 8 % 1.6 = 1.5999999999999996
+        # 5.4 % 1.8 = 2.220446049250313e-16
+        if (
+            product_qty
+            and packaging_qty
+            and float_compare(
+                product_qty / packaging_qty,
+                float_round(product_qty / packaging_qty, precision_rounding=1.0),
+                precision_rounding=default_uom.rounding
+            )
+            != 0
+        ):
```

Comment:
it's not needed I think since qty should be different than zero it should not be able to divide by zero (except with big uom difference that could be the only check)

Reply:
I don't get it. This code is to check if the result `product_qty / packaging_qty` is a "integer". Only if the result is not a "integer", so `product_qty` is not a multiple of `packaging_qty`, we round `product_qty` to a multiple of `packaging_qty`.

Reply:
@yhu-odoo It's just that any case you are doing the float_round to check if it's similar or not so it would be faster to always
return float_round(product_qty / packaging_qty, precision_rounding=1.0, rounding_method=rounding_method) * packaging_qty
Since if it's equal it will return the same qty than product_qty and if it's not a multiple it will round the closer multiple according rounding_method

Reply:
@amoyaux 
I thought about this before, but if we directly return float_round, then we didn't take care any precision_rounding.
for example, we may have product_qty / packaging_qty = 2.99 which we don't want to calculate a new qty. Now, after we do 3 * packaging_qty, the rounding error may get big enough to make the new qty different from our original one.

Reply:
Okay even if I think it's okay to return a new value since if you have precision now, you will also probably have them in the remaining of the process